src/shared/sleep-config.c

```diff
@@ -259,6 +265,52 @@ static bool enough_memory_for_hibernation(void) {
         return r;
 }
 
+static int parse_proc_cmdline_item(const char *key, const char *value) {
+
+        assert(key);
+
+        if (streq(key, "resume") && value) {
+                free(arg_resume_dev);
+                arg_resume_dev = fstab_node_to_udev_node(value);
+                if (!arg_resume_dev) {
+                        return log_oom();
+                }
+        }
+
+        return 0;
+
+}
+
+static bool resume_passed_to_kernel(void) {
+        int r = 0;
+        struct stat rd;
+
+        /* If not using the systemd hibernate generator we don't care about the cmdline */
+        if (!validate_hibernate_generator)
+                return true;
```

Comment:
That doesn't solve the issue of no resume= present but the systemd hibernation generator is in use (eg Fedora's config) and logind reporting CanHibernate as 'yes' 

The whole point of the patch is for distributions using the systemd hibernate generator to validate resume= is present when determining if 'yes' or 'na' should be reported on DBus to the CanHibernate call.

If no resume= meant the logic was skipped ... well it'd defeat the purpose of the patch.

And if not using the hibernate generator this would be the correct logic - don't bother checking resume= for CanHibernate as alternative methods are being used.